### ISSUE TITLE
How to use Astro's <Image/> component with the <Card/> component?

### ISSUE THREAD

**Issue by jasonday** [incluud/accessible-astro-components#107]
I'm using the Accessibility Starter, and in `[..page].astro` there's the following:
```
 <ul class="my-3">
        {
          page.data.map((post) => (
            <li>
              <Card
                img={`${post.data.image}`}
                url={'/blog/' + post.id}
                title={post.data.title} 
                footer={'Author: ' + post.data.author}
              >
                {post.data.description}
              </Card>
            </li>
          ))
        }
      </ul>
```

And in my .mdx file:
```
image: {robot}
imageAlt: "A broken robot, electricity sparking and smoking, saying 'install me'"
---

import { Picture, Image } from 'astro:assets';
import robot from '../../assets/img/2025/robot-widget.png';
```

How do I properly have `<Card/>` utilize Astro's `<Image/>` helper?

**Comment by jasonday** [comment]
For anyone else that lands here, I copied `Card` into a new component `ImageCard` and set it to use Astro's `<Image/>` helper. 

Code:
https://gist.github.com/jasonday/7239eb243efd70cbc65448b0f4a5ed1c

**Comment by markteekman** [comment]
Hey @jasonday,

Thank you for providing an alternative to use Astro's `<Image />` component! It's definitely something we can consider adding as an option for the Card component. A small concern I should note in your `<ImageCard />` component: if you're applying an `alt` text to the image you should also change the order of the image and the title in your HTML structure. According to WCAG 2.2 [Success Criterion 1.3.2 Meaningful Sequence (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/meaningful-sequence.html), the reading order should be logical and meaningful. In your current situation, for a none sighted user, the alt text of the image would be read before the title, which could lead to confusion. So always make sure information is preceded by the title of the card. You could change the HTML structure and use `order: -1;` like we do with the meta tags for example to fix this. Just remember to always be mindful when using alt texts in these kind of situations and consider:

When to add it:

- Image contains unique data/information not in the text
- Image shows critical visual details central to understanding
- Image provides essential context missing from the text
- Image contains text not present elsewhere
- Image is the primary content being showcased

Potential downsides:

- Creates redundancy if repeating information already in title/description
- Can lead to inconsistent implementation across pages
- May cause cognitive overload with too much information
- Can create reading order confusion if DOM structure isn't optimal
- Adds maintenance burden as images change
- May create misaligned experiences between user groups

Like I mentioned in #33, if you have any information or resources about this do let me know! I always aim to improve my knowledge about accessibility to provide the best possible experience for people using this library and potential end-users 🙂 I'm discussing this in a Slack group with other accessibility experts and based on the conclusions there I'm going to see if we can provide the alt prop as well, and document on how and when to use it.

Thanks in advance!

**Comment by markteekman** [comment]
Hey @jasonday,

Hmm, interesting 🤔 I agree with you that chancing the order violates Technique C27. I've had different people (mostly auditors or accessibility experts) tell me different solutions when it comes to matters like these and it's frustrating that there are so many different opinions about it. It's exactly why I like talking about it with as much people who've got experience in the field, like you do. 

I'll move this issue to our roadmap for now and I'll let you know when I hear back from our Slack community. Thanks so far!

**Comment by markteekman** [comment]
@jasonday I have talked to some people on the accessibility Slack channel I mentioned and here's what I got from it:

- Images in cards can be either informative or decorative depending on context
- Component should allow developers to specify whether an image is decorative or informative
- One expert's view evolved to favor providing alt text even for seemingly decorative images:
  - Low vision users may perceive something is there but can't distinguish it
  - Providing simple descriptions gives users the choice to decide if it's valuable
  - Better to provide information than make assumptions about what users need
- Ideal HTML order places the heading first, then the image
- DOM order matching visual order has benefits but may be less critical in some contexts
- If a card has only link text and a decorative image, you might create a heading with no content below it
- Visual order can be adjusted using CSS grid and flexbox without changing DOM order
- The common approach of using `::after` to make entire cards clickable may prevent Fusion users from accessing the alt text

### Conclusion
Based on this feedback, including an option for alt text while being mindful of DOM order appears to be the most inclusive approach, allowing for context-specific decisions on whether images are decorative or informative. For now, we'll provide a way to set the alt text or have a `<slot>` to use Astro's `<Image />` component instead. I'll take these changes into account when we do a next iteration of improvements 🙂 

Thanks again!

**Comment by markteekman** [comment]
@jasonday just released an update today and it's now possible to use the `<Image />` component directly in the `<Card>` as an alternative to the default `<img>`! 🥳 You can also provide an alt text in both cases.  I still have to update the docs and an update to the Accessible Astro Starter is almost done which has the blog and portfolio both using this approach. Haven't included the DOM order part yet, but I'll look at that another time. For now you can use it like this:

```astro
---
import myImage from '@assets/images/myImage.png'
---

<Card 
  title="Card Title"
  imageComponent={myImage}
  imageAlt="Description of image"
/>
```

Thanks again for opening the issue!
